### PR TITLE
Create tag on reconciler

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind ci

#### What this PR does / why we need it:
The tag we try to push is not being created beforehand, which is now fixed when applying this patch.

#### Which issue(s) this PR fixes:

None
#### Special notes for your reviewer:
None
#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Comment by saschagrunert** [comment]
cc @olad5

File: scripts/tag-reconciler/tag-reconciler.go [hunk 1]
```diff
@@ -65,6 +65,11 @@ func run() error {
 }
 
 func pushTagToRemote(repo *git.Repo, tag, remote string) error {
+	logrus.Info("Tagging repository")
```
Comment (kwilczynski) [inline scripts/tag-reconciler/tag-reconciler.go:68]:
```suggestion
	logrus.Infof("Adding tag to repository: %s", tag)
```

File: scripts/tag-reconciler/tag-reconciler.go [hunk 2]
```diff
@@ -65,6 +65,11 @@ func run() error {
...
+	if err := repo.Tag(tag, tag); err != nil {
+		return fmt.Errorf("tag repository: %w", err)
```
Comment (kwilczynski) [inline scripts/tag-reconciler/tag-reconciler.go:70]:
```suggestion
		return fmt.Errorf("unable to tag repository: %w", err)
```

File: scripts/tag-reconciler/tag-reconciler.go (hunk 1 above)
Comment (saschagrunert) [reply scripts/tag-reconciler/tag-reconciler.go]: Applied as suggested.

File: scripts/tag-reconciler/tag-reconciler.go (hunk 2 above)
Comment (saschagrunert) [reply scripts/tag-reconciler/tag-reconciler.go]: Applied as suggested.

**Pull request merged**